src/databricks/labs/ucx/framework/parallel.py

```diff
@@ -45,22 +48,36 @@ def __init__(
     @classmethod
     def gather(cls, name: str, tasks: list[ExecutableFunction]) -> list[ExecutableResult]:
         reporter = ProgressReporter(len(tasks), f"{name}: ")
-        return cls(tasks, num_threads=4, progress_reporter=reporter).run()
+        return cls(name, tasks, num_threads=4, progress_reporter=reporter).run()
 
     def run(self) -> list[ExecutableResult]:
         logger.debug(f"Starting {len(self._executables)} tasks in {self._num_threads} threads")
 
+        def error_handler(func, name):
+            @functools.wraps(func)
+            def inner(*args, **kwargs):
+                try:
+                    return func(*args, **kwargs)
+                except Exception as ex:
+                    msg = getattr(ex, "message", ex)
+                    logger.warning(f"An error occurred while '{name}', message: {msg}")
+                    return None
+
+            return inner
+
         with ThreadPoolExecutor(self._num_threads) as executor:
             for executable in self._executables:
-                future = executor.submit(executable)
+                future = executor.submit(error_handler(executable, self._name))
                 future.add_done_callback(self._done_callback)
-                # TODO: errors are not handled yet - https://github.com/databricks/UC-Upgrade/issues/89
                 self._futures.append(future)
 
             results = concurrent.futures.wait(self._futures, return_when=ALL_COMPLETED)
 
         logger.debug("Collecting the results from threaded execution")
-        collected = [future.result() for future in results.done]
+        collected = [future.result() for future in results.done if future.result()]
```

Comment:
i think there's a bug here, as `.result()` may not always be available the second time we call it. Just for counting, introduce a separate variable.

Reply:
Really? I thought (by reading the `_base.py` code) that only reading exceptions has side effects?